internal/clusterconfig/clusterconfig.go

```diff
@@ -168,39 +161,17 @@ func (r *UpgradeClusterConfigGather) fetchProxy(ctx context.Context, manifestsDi
 	return utils.MarshalToFile(p, filePath)
 }
 
-func (r *UpgradeClusterConfigGather) fetchMachineConfigs(ctx context.Context, manifestsDir string) error {
-	for _, machineConfigName := range machineConfigNames {
-		r.Log.Info("Fetching MachineConfig", "name", machineConfigName)
-
-		machineConfig := mcv1.MachineConfig{}
-		if err := r.Client.Get(ctx, types.NamespacedName{Name: machineConfigName}, &machineConfig); err != nil {
-			return err
-		}
-
-		mc := mcv1.MachineConfig{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:   machineConfig.Name,
-				Labels: machineConfig.Labels,
-			},
-			Spec: machineConfig.Spec,
-		}
-		machineConfigTypeMeta, err := r.typeMetaForObject(&machineConfig)
-		if err != nil {
-			return err
-		}
-		mc.TypeMeta = *machineConfigTypeMeta
-
-		filePath := filepath.Join(manifestsDir, machineConfig.Name+".json")
-		r.Log.Info("Writing MachineConfig to file", "path", filePath)
-		if err := utils.MarshalToFile(mc, filePath); err != nil {
-			return err
-		}
+func (r *UpgradeClusterConfigGather) fetchSSHPublicKey() (string, error) {
+	sshKey, err := os.ReadFile(filepath.Join(hostPath, sshKeyFile))
```

Comment:
There might be multiple keys in that file

Reply:
Yep and this is fine, it will write it as string and will actually help us to copy all the ssh keys